services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -55,9 +56,13 @@ export const ChoiceListField = ({
     if (displayValue) {
       form.setValue(name, displayValue, { shouldValidate: true });
     }
+
     // update DOM choices checked status
     choices.forEach((choice: FieldChoice) => {
-      choice.checked = displayValue?.includes(choice.value) || false;
+      const foundValue = displayValue?.find(
+        (value) => value.value === choice.value
+      );
+      choice.checked = !!foundValue;
```

Comment:
Nit: I found this method a little hard to parse due to all the `value`. Maybe:
```suggestion
      const checkedState = displayValue?.find(
        (option) => option.value === choice.value
      );
      choice.checked = !!checkedState;
```